### PR TITLE
fix(Select): Fix accessibility issue. 

### PR DESCRIPTION
**Pull request by nikitaorliak-cengage**
Issue: #1390

## What I did

Remove the `ClearIndicator` component outside from the `SelectTriggerButton`.

## Checklist 
- [x] changeset has been added
- [x] Pull request description is descriptive
- [ ] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by silvalaura** [comment]
This looks good, and I verified the screenreader reads it on Safari.

However, because we are positioning it absolutely, it breaks in the examples with `labelPosition="left"` or when `isLabelVisuallyHidden` is true:
image
image
image

**Comment by orion-cengage** [comment]
We need to stop the clear button from overlapping the text and basically disappearing.

image

**Comment by orion-cengage** [comment]
Not sure when this became an issue, but the regular Select is not supposed to grow vertically just because an item is long and wraps.

image

If anything, we should truncate it with ellipsis like this:

image

**Comment by orion-cengage** [comment]
The positioning of the clear button is off when it is in the error state or when showing helper text.

image

image

**Comment by orion-cengage** [comment]
The inverse version of the input isn't using the inverse version of the clear button so it is hard to see.

image

**Review by orion-cengage — CHANGES_REQUESTED** [review]
I added a number of styling issues.

**Review by orion-cengage — CHANGES_REQUESTED** [review]
The latest update fixed the previous styling issues but has broken the focus style on the main container of the Select input. If we can fix this, then we might be good to go.

image

image

image

**Comment by silvalaura** [comment]
It's so close! 
- Focus is back ✅ 
- Text overflow is properly cut off ✅ 
- X is properly aligned for Combobox with left label ✅
- - X is properly aligned for Combobox with top label ✅ 
- X is properly aligned for Select with left label ✅ 
- X is properly aligned for Select with top label ❌ 
- Is inverse looks good ✅

**Comment by silvalaura** [comment]
One more:
if Select `isMulti` is set to true, the isClearable button disappears. (it is visible for Combobox `isMulti`)
image

All other use cases look good to me.

File: packages/react-magma-dom/src/components/Select/MultiSelect.tsx [hunk 1]
```diff
@@ -159,6 +163,44 @@ export function MultiSelect<T>(props: MultiSelectProps<T>) {
...
+        allItems.push(item.label);
+      }
+    });
+
+    return allItems.join(', ');
+  }
+
+  const toggleButtonRef = React.useRef<HTMLButtonElement>();
+
+  const clearIndicatori18n =
```
Comment (silvalaura) [inline packages/react-magma-dom/src/components/Select/MultiSelect.tsx:185]: In this multiselect file, wouldn't we always want to use the multi label?

**Review by silvalaura — COMMENTED** [review]
**Select:**
- Focus around the entire input in Select ✅ 
- Focus is only on isClearable (no double focus) ✅
- Text overflow is properly cut off ⚠️ **This looks good, but would be even better if we could add like 8px of space between the ellipsis and the clear button so they never overlap at all**
image
- X is properly aligned for Select with left label ✅
- X is properly aligned for Select with top label ✅
- X is properly aligned for Multi Select with left label ⚠️ **How hard would it be to align to the center? It's center aligned for Combobox**
image
- X is properly aligned for Multi Select with top label ⚠️ **It can get in this weird state where the X is over the text. Maybe if we add the space noted above, this could look even better**
image 
- isInverse uses correct colors for Select ✅
- Screenreader on Safari reads isClearable ✅
- Screenreader on Chrome reads isClearable ✅

**Combobox:**
- Focus around the entire input in Combobox ✅
- Focus is only on isClearable ✅
- Text overflow is properly cut off ✅
- X is properly aligned for Combobox with left label ✅
- X is properly aligned for Combobox with top label ✅
- X is properly aligned for Multi Combobox with left label ✅
- X is properly aligned for Multi Combobox with top label ✅
- isInverse uses correct colors for Combobox ✅
- Screenreader on Safari reads isClearable ✅
- Screenreader on Chrome reads isClearable ✅

**Comment by silvalaura** [comment]
We're almost there!

**Select:**
- Focus around the entire input in Select ✅ 
- Focus is only on isClearable (no double focus) ⚠️ **can it be moved _slightly_ more to the left to avoid the focus touching the arrow?**
image
- Text overflow is properly cut off ✅
- X is properly aligned for Select with left label ✅
- X is properly aligned for Select with top label ✅
- X is properly aligned for Multi Select with left label ✅
- X is properly aligned for Multi Select with top label ✅
- isInverse uses correct colors for Select ✅
- Screenreader on Safari reads isClearable ✅
- Screenreader on Chrome reads isClearable ✅

**Combobox:**
- Focus around the entire input in Combobox ✅
- Focus is only on isClearable ✅
- Text overflow is properly cut off ✅  I noticed the issue below, but I can see it's an existing issue, so it does not need to be addressed at this time. 
image
image
- X is properly aligned for Combobox with left label ✅
- X is properly aligned for Combobox with top label ✅
- X is properly aligned for Multi Combobox with left label ✅
- X is properly aligned for Multi Combobox with top label ✅
- isInverse uses correct colors for Combobox ✅
- Screenreader on Safari reads isClearable ✅
- Screenreader on Chrome reads isClearable ✅

**Review by orion-cengage — APPROVED** [review]

**Review by silvalaura — APPROVED** [review]

**Pull request merged**